### DISCUSSION TITLE
Using private dependencies based on bitbucket repository SSH url

### DISCUSSION THREAD

**Discussion by burekas7** [codemagic-ci-cd/codemagic-docs#2380] [Q&A]
I have the same issue as described here:
https://stackoverflow.com/questions/75591706/using-private-dependencies-with-codemagic

Following this guide:
https://docs.codemagic.io/knowledge-git/access-private-git-submodules/

I'm using the Workflow editor.
How can I add a group? Why is it needed?

### Which mobile framework are you using?

Flutter (Dart)

### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

**Comment by dtrdic** [comment]
Hi @burekas7, variable group name option is only available if you use yaml configuration (you can't combine yaml configuration and workflow editor configuration).
If you are working with workflow editor then the variable group name is not needed, just add the variable without it
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/223f5b6c-4013-4afc-b5dc-af33cbf169e8)

**Comment by burekas7** [comment]
@dtrdic 
Thanks.
I have another question regarding Bitbucket and SSH key.
If I integrated an app with Bitbucket repository, and in addition I have a private dependencie inside the app that based on another repository.
(Both repositories are in the same place)
Do I need to add an SSH key also as enviroment variable or I don't need to?

**Reply by dtrdic** [reply]
Note: responded in Discord channel, but in case you haven't see I'll ask here as well..
How did you add your repository to codemagic, via SSH or OAuth? Are you trying to build your app on windows machine? Please also share a build id so i can take a closer look

**Reply by burekas7** [reply]
I think it was using SSH, I don't remember, 
I just followed the wizard from the Team/Personal account integrations section.
The app is built using Mac machine.

Where can I find the build id?
I haven't tried it yet building an app using the private dependencie.

**Reply by burekas7** [reply]
@dtrdic 

<img width="1309" alt="Screenshot 2024-03-03 at 11 11 28" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/13424257/267f8834-8e79-497f-ae54-6fe05c254a6e">

**Reply by dtrdic** [reply]
It can be found on the build overview page once you start a build. However, since you mentioned that you haven't tried to run a build yet, I cannot verify if there are any other errors. In any case, if your repository contains private dependencies, you will need to add the SSH key as an environment variable, as I mentioned earlier. Once you initiate a build, please share the build ID with me if there are any error messages so that I can investigate further

**Reply by dtrdic** [reply]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion

**Reply by burekas7** [reply]
@dtrdic 

I have problems with my private dependency when building an app:
ID: 65ff42bd35b9e3033aad73c2

The _SSH_KEY was added to the app's Environment Variables.
* My App also based on another Bitbucket repository.
 
<img width="843" alt="Screenshot 2024-03-23 at 23 34 18" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/13424257/79bc7bfe-27ae-4abf-8d0b-ce9b2baae0f9">
